src/components/Settings/MaxSlippageSettings/index.tsx

```diff
@@ -0,0 +1,169 @@
+      if (parsed > 5000) {
+        setSlippageError(SlippageError.InvalidInput)
+      } else {
+        setUserSlippageTolerance(new Percent(parsed, 10_000))
+      }
+    } catch (e) {
+      setSlippageError(SlippageError.InvalidInput)
+    }
+  }
+
+  const tooLow =
+    userSlippageTolerance !== SlippageTolerance.Auto && userSlippageTolerance.lessThan(MINIMUM_RECOMMENDED_SLIPPAGE)
+  const tooHigh =
+    userSlippageTolerance !== SlippageTolerance.Auto && userSlippageTolerance.greaterThan(MAXIMUM_RECOMMENDED_SLIPPAGE)
+
+  return (
+    <Expand
+      header={
+        <Row width="auto">
+          <ThemedText.BodySecondary>
+            <Trans>Max slippage</Trans>
+          </ThemedText.BodySecondary>
+          <QuestionHelper
+            text={
+              <Trans>Your transaction will revert if the price changes unfavorably by more than this percentage.</Trans>
+            }
+          />
+        </Row>
+      }
+      button={
+        <ThemedText.BodyPrimary>
+          {userSlippageTolerance === SlippageTolerance.Auto ? (
+            <Trans>Auto</Trans>
+          ) : (
+            `${userSlippageTolerance.toFixed(2)}%`
+          )}
+        </ThemedText.BodyPrimary>
+      }
+    >
+      <RowBetween gap="md">
+        <Switch>
+          <Option
+            onClick={() => {
+              // Reset the input field when switching to auto
+              setSlippageInput('')
+              setUserSlippageTolerance(SlippageTolerance.Auto)
+            }}
+            isActive={userSlippageTolerance === SlippageTolerance.Auto}
+          >
+            <ThemedText.BodyPrimary>
+              <Trans>Auto</Trans>
+            </ThemedText.BodyPrimary>
+          </Option>
+          <Option
+            onClick={() => {
+              // When switching to custom slippage, use `auto` value as a default.
+              setUserSlippageTolerance(autoSlippage)
+            }}
+            isActive={userSlippageTolerance !== SlippageTolerance.Auto}
+          >
+            <ThemedText.BodyPrimary>
+              <Trans>Custom</Trans>
+            </ThemedText.BodyPrimary>
+          </Option>
+        </Switch>
+        <InputContainer gap="md" error={!!slippageError}>
+          <Input
+            placeholder={autoSlippage.toFixed(2)}
+            value={slippageInput}
+            onChange={(e) => parseSlippageInput(e.target.value)}
+            onBlur={() => {
+              // When the input field is blurred, reset the input field to the current slippage tolerance
+              setSlippageInput(userSlippageTolerance !== SlippageTolerance.Auto ? userSlippageTolerance.toFixed(2) : '')
+              setSlippageError(false)
+            }}
+          />
+          <ThemedText.BodyPrimary>%</ThemedText.BodyPrimary>
+        </InputContainer>
+      </RowBetween>
+      {tooLow || tooHigh ? (
+        <RowBetween
+          style={{
+            fontSize: '14px',
+            paddingTop: '7px',
```

Comment:
7px seems wrong, should we round it to 8? also, i think we wanted to avoid custom `style` props and use styled components for everything.